src/controller/RentalsController.java

```diff
@@ -126,13 +132,18 @@ public static Rental getRental(int rid) {
 
         //TODO: This query is incorrect. It does not do what is expected
         String query = "SELECT * ";
```

Comment:
Could you add to the initial query the whole query until "AND R.vlicense = V.vlicense"? It seems like it's the same for both cases. The only difference is the location and city stuff and the Group By